src/operator/race-support.ts

```diff
@@ -0,0 +1,61 @@
+import {Operator} from '../Operator';
+import {Observable} from '../Observable';
+import {Subscriber} from '../Subscriber';
+import {OuterSubscriber} from '../OuterSubscriber';
+import {subscribeToResult} from '../util/subscribeToResult';
+
+export class RaceOperator<T, R> implements Operator<T, R> {
+  call(subscriber: Subscriber<T>): Subscriber<T> {
+    return new RaceSubscriber(subscriber);
+  }
+}
+
+export class RaceSubscriber<T, R> extends OuterSubscriber<T, R> {
+  private active: number = 0;
+  private hasFirst: boolean = false;
+  private first: number;
+  private observables: Observable<any>[] = [];
+
+  constructor(destination: Subscriber<T>) {
+    super(destination);
+  }
+
+  _next(observable: any): void {
+    this.observables.push(observable);
+  }
+
+  _complete() {
+    const observables = this.observables;
+    const len = observables.length;
+    if (len === 0) {
+      this.destination.complete();
+    } else {
+      this.active = len;
+      for (let i = 0; i < len; i++) {
+        const observable = observables[i];
+        this.add(subscribeToResult(this, observable, observable, i));
+      }
+    }
+  }
+
+  notifyComplete(unused: Subscriber<R>): void {
+    if (!this.hasFirst) {
+      this.destination.complete();
+    } else {
+      if ((this.active -= 1) === 0) {
+        this.destination.complete();
+      }
+    }
+  }
+
+  notifyNext(observable: any, value: R, outerIndex: number): void {
+    if (!this.hasFirst) {
+      this.hasFirst = true;
+      this.first = outerIndex;
```

Comment:
I could try making it this way. I originally tried but didn't really know how to make eager unsubscription worked, since I was trying to call the method but it was failing in the unit tests. Sounds reasonable to unsub the rest instead of just keeping them like the other combination ops.